ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/VertxNonceRequestTransmitter.java

```diff
@@ -78,32 +74,37 @@ public BigInteger requestNonce(final JsonRpcRequest request) {
   private CompletableFuture<BigInteger> getNonceFromWeb3Provider(
       final JsonRpcRequest requestBody, final MultiMap headers) {
 
-    requestBody.setId(new JsonRpcRequestId(nextId.getAndIncrement()));
-
     final CompletableFuture<BigInteger> result = new CompletableFuture<>();
 
-    final HttpClientRequest request =
-        client.request(
-            HttpMethod.POST,
-            downstreamPathCalculator.calculateDownstreamPath("/"),
-            response -> response.bodyHandler(responseBody -> handleResponse(responseBody, result)));
-
-    request.setTimeout(requestTimeout.toMillis());
-    request.headers().setAll(headers);
-    request.exceptionHandler(result::completeExceptionally);
-    request.headers().remove("Content-Length"); // created during 'end'.
-    request.setChunked(false);
-    request.end(Json.encode(requestBody));
+    final RequestTransmitter transmitter =
+        transmitterFactory.create(
+            new DownstreamResponseHandler() {
```

Comment:
Think it would be clearer to make this an inner class instead of a anonymous class